libraries/c_sdk/standard/mqtt/src/private/iot_mqtt_internal.h

```diff
@@ -251,12 +251,21 @@
 /**
  * @brief Default config for Maximum Number of MQTT CONNECTIONS.
  * This config can be specified by the application based on number of MQTT
- * connections needed.
+ * connections needed. Maximum number of subscriptions that shim can hold is 2^7.
  */
 #ifndef MAX_NO_OF_MQTT_CONNECTIONS
     #define MAX_NO_OF_MQTT_CONNECTIONS    ( 2 )
 #endif
 
+/**
+ * @brief Default config for Maximum Number of MQTT Subscriptions.
+ * This config can be specified by the application based on number of MQTT
+ * subscriptions.
+ */
+#ifndef MAX_NO_OF_MQTT_SUBSCRIPTIONS
+    #define MAX_NO_OF_MQTT_SUBSCRIPTIONS    ( 10 )
```

Comment:
Do we need the default value to be a bit high, how about 20 or 30?

Reply:
Demos and tests were working fine with this value, so I kept it this. Also it can always be configured by the application.

Reply:
Ok. I think keeping it 10 as we have tested Shadow and defender demo is fine, but I think we would need to capture the need of this config very clearly in our public documentation, not in the scope of this PR.